sound-design.MD

```diff
@@ -0,0 +1,42 @@
+# MRE Sound API proposal
+
+This is node.js side only. It doesn't actually do anything yet.
+
+## Required functionality
+* preload sounds (as looping or one-shot)
+* play preloaded sounds
+* load and play sounds (Basically preload, instantly play, unload upon completion)
+* pause, resume, 
+* stop sounds 
+* set the sound state (to change volume, pitch, and where possible seek)
+
+## Design thoughts
+
+(Feel free to comment on each item separately)
+
+I'm using assetmanager to for the preloading sound to ask the client to load a sound resource, and return a handle. This is therefore are not instant, but return a promise.
+
+All sounds play on an actor, and the play calls therefore happen on the actor.
+
+PlaySound are played by passing in a sound resource ID, and instantly return a sound instance ID.
+
+LoadAndPlaySound takes a URI, and returns a promise to a sound instance ID when the primary client has started playing the sound.
+
+The sound instance doesn't store any data on the server at all, it lets the client manage the instance fully. It's up to the client to take the sound instance handle and tweak it if necessary. Sound instance handles expire the moment a non-looping sound reaches the end, or (for both looping/non-looping sounds) when the user calls stop() for that sound instance. 
```

Comment:
>ets the client manage the instance fully [](start = 68, length = 40)

This would make sounds heavily dependent on the authoritative peer. Seems like it would make sense to at least save clip duration and start time server-side, so we can estimate completion time and auto-expire instances without an extra round trip.

Reply:
I don't think the server needs to know the length of the clip. It just needs to know when it started playing and when has stopped. The authoritative client would need to send a message to let the server know the sound has ended.

---
In reply to: [256945051](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/191#discussion_r256945051) [](ancestors = 256945051)

Reply:
Yeah, I'm changing the design here to have it managed by the server instead. And yeah, it will be relying on stop messages

---
In reply to: [257022732](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/191#discussion_r257022732) [](ancestors = 257022732,256945051)